presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -1008,4 +1010,24 @@ public void testSTGeometryType()
         assertFunction("ST_GeometryType(ST_GeometryFromText('GEOMETRYCOLLECTION(POINT(4 6),LINESTRING(4 6, 7 10))'))", VARCHAR, "ST_GeomCollection");
         assertFunction("ST_GeometryType(ST_Envelope(ST_GeometryFromText('LINESTRING (1 1, 2 2)')))", VARCHAR, "ST_Polygon");
     }
+
+    private void assertToFromBinary(String wkt)
+    {
+        assertToFromBinary(wkt, wkt);
+    }
+
+    private void assertToFromBinary(String wkt, String expected)
+    {
+        assertFunction(format("ST_AsText(ST_GeometryFromBinary(ST_AsBinary(ST_GeometryFromText('%s'))))", wkt), VARCHAR, expected);
+    }
+
+    @Test
+    public void testSTGeometryFromBinary()
+    {
+        assertToFromBinary("POINT EMPTY");
+        assertToFromBinary("LINESTRING (1 1, 2 2)");
+        assertToFromBinary("POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))");
+        assertToFromBinary("POLYGON ((1 1, 1 4, 4 4, 4 1))", "POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))");
```

Comment:
Why would these be different?

Reply:
Apparently, this is expected behavior. See https://github.com/prestodb/presto/blob/master/presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java#L84.

Reply:
@mpertierra Marcos, in the canonical representation the vertexes are listed in counterclockwise order. In this test case, the vertexes are listed in clockwise order. `assertToFromBinary("POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))"` should work.

The other test needs to be modified as well. If you could do that in a separate commit, that would be appreciated.